planner/core/planbuilder.go

```diff
@@ -1615,7 +1615,12 @@ func buildShowSchema(s *ast.ShowStmt) (schema *expression.Schema) {
 	case ast.ShowCreateDatabase:
 		names = []string{"Database", "Create Database"}
 	case ast.ShowGrants:
-		names = []string{fmt.Sprintf("Grants for %s", s.User)}
+		if s.User != nil {
+			names = []string{fmt.Sprintf("Grants for %s", s.User)}
+		} else {
+			// Don't know the name yet, so just say "user"
+			names = []string{fmt.Sprintf("Grants for User")}
```

Comment:
we can save the function call of `fmt.Sprintf()`, use the string "Grants for User" instead.